### PR TITLE
split up a few files in src/basic/

### PR DESCRIPTION
**Pull request by poettering**
let's do some spring cleaning in october and split some bits out of existing header files, to make them a bit shorter, individually.

No real code changes, just shifting stuff around

(ok, admittedly, I replaced some redundant code with calls to sync_full(), but that's so trivial a clean-up that it shouldn't matter)

### CODE REVIEW

File: src/shared/format-table.c [hunk 1]
```diff
@@ -13,9 +13,9 @@
 #include "format-util.h"
 #include "fs-util.h"
 #include "gunicode.h"
+#include "glyph-util.h"
```
Comment (yuwata) [inline src/shared/format-table.c]: Not sorted.

File: src/partition/repart.c [hunk 2]
```diff
@@ -34,6 +35,7 @@
 #include "format-table.h"
 #include "format-util.h"
 #include "fs-util.h"
+#include "fs-util.h"
```
Comment (yuwata) [inline src/partition/repart.c]: Duplicated??

**Review by yuwata — CHANGES_REQUESTED** [review]
LGTM. But two nits.

**Comment by mrc0mmand** [comment]
CI no likey:

```
../src/shared/user-record.c: In function ‘read_login_defs’:
../src/shared/user-record.c:57:13: error: implicit declaration of function ‘chase_symlinks_and_fopen_unlocked’ [-Werror=implicit-function-declaration]
   57 |         r = chase_symlinks_and_fopen_unlocked(path, root, CHASE_PREFIX_ROOT, "re", &f, NULL);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/shared/user-record.c:57:13: error: nested extern declaration of ‘chase_symlinks_and_fopen_unlocked’ [-Werror=nested-externs]
../src/shared/user-record.c:57:59: error: ‘CHASE_PREFIX_ROOT’ undeclared (first use in this function)
   57 |         r = chase_symlinks_and_fopen_unlocked(path, root, CHASE_PREFIX_ROOT, "re", &f, NULL);
      |                                                           ^~~~~~~~~~~~~~~~~
../src/shared/user-record.c:57:59: note: each undeclared identifier is reported only once for each function it appears in
cc1: all warnings being treated as errors
```

**Comment by yuwata** [comment]
```
../src/shared/user-record.c: In function â€˜read_login_defsâ€™:
../src/shared/user-record.c:58:84: error: passing argument 5 of â€˜chase_symlinks_and_fopen_unlockedâ€™ from incompatible pointer type [-Werror=incompatible-pointer-types]
   58 |         r = chase_symlinks_and_fopen_unlocked(path, root, CHASE_PREFIX_ROOT, "re", &f, NULL);
      |                                                                                    ^~
      |                                                                                    |
      |                                                                                    FILE **
In file included from ../src/shared/user-record.c:6:
../src/basic/chase-symlinks.h:32:128: note: expected â€˜char **â€™ but argument is of type â€˜FILE **â€™
   32 | int chase_symlinks_and_fopen_unlocked(const char *path, const char *root, unsigned chase_flags, const char *open_flags, char **ret_path, FILE **ret_file);
      |                                                                                                                         ~~~~~~~^~~~~~~~
cc1: all warnings being treated as errors
```

**Comment by bluca** [comment]
I did a no-change force push, to trigger github actions, since they are now back online after the outage

**Pull request merged**